docs/encyclopedia/application-message-passing.mdx

```diff
@@ -26,57 +28,85 @@ keywords:
 - temporal sdk message passing
 ---
 
+import PrettyImage from '@site/src/components/pretty-image/PrettyImage';
 import { RelatedReadContainer, RelatedReadItem } from '@site/src/components/related-read/RelatedRead';
 
-Temporal applications can receive and provide data through messages.
+Workflows can be thought of as stateful web services that can receive messages. The Workflow can have powerful message handlers akin to endpoints that can react to the incoming messages in combination with the current state of the Workflow.
 Temporal supports three types of messages: Signals, Queries, and Updates.
+To define these, consider these messages from the perspective of the client making the request:
 
-## Signals {#signals}
+- Queries are read requests. They can read the current state of the Workflow but cannot block in doing so.
+- Signals are write requests. They cause changes in the running Workflow, but you cannot await any response or error.
+- Updates are interactive write requests, wherein the sender of the Update tracks completion and waits for a response or an error.
```

Comment:
I don't think "interactive" is quite the right word here; how about

> Updates are trackable write requests

which I think you had in an earlier version.


"Interactive" suggests a bit too much, i.e. that the client can modify the execution of the request after first issuing it. An "interactive request" is not currently a standard concept in software engineering and in our case, we don't even support cancellation of update requests yet.

Reply:
I changed to "synchronous trackable" which is what I was going for with "interactive".